### PR TITLE
docs: Update image.rst

### PR DESCRIPTION
**Pull request by microbit-mark**
update as per https://github.com/bbcmicrobit/micropython/issues/437
- describe ways to construct image
- note that buffer will not accept kwargs
- provide example of buffer image

### CODE REVIEW

**Comment by microbit-mark** [comment]
@carlosperate for review

File: docs/image.rst [hunk 1]
```diff
@@ -11,6 +11,17 @@ the ``display`` API::
 
 .. image:: image-smile.png
 
+There are four ways in which you can construct an image:
+
+Image() - create a blank 5x5 image
```
Comment (microbit-carlos) [inline docs/image.rst]: Trying out the new GitHub feature to suggest patches to the PR:

```suggestion
- ``Image()`` - Create a blank 5x5 image
```

File: docs/image.rst [hunk 2]
```diff
@@ -11,6 +11,17 @@ the ``display`` API::
...
+
+Image(s) - create an image by parsing the string (single char returns that glyph)
```
Comment (microbit-carlos) [inline docs/image.rst]:
```suggestion
- ``Image(string)`` - Create an image by parsing the string, a single character
  returns that glyph
```

File: docs/image.rst [hunk 3]
```diff
@@ -11,6 +11,17 @@ the ``display`` API::
...
+
+Image(w, h) - create a blank image of given size
```
Comment (microbit-carlos) [inline docs/image.rst]:
```suggestion
- ``Image(width, height)`` - Create a blank image of given size
```

File: docs/image.rst [hunk 4]
```diff
@@ -11,6 +11,17 @@ the ``display`` API::
...
+
+Image(w, h, data) - create an image from the given data
```
Comment (microbit-carlos) [inline docs/image.rst]:
```suggestion
- ``Image(width, height, buffer)`` - Create an image from the given buffer
```

File: docs/image.rst [hunk 5]
```diff
@@ -39,8 +50,15 @@ Classes
 
     The other form creates an empty image with ``width`` columns and
     ``height`` rows. Optionally ``buffer`` can be an array of
-    ``width``×``height`` integers in range 0-9 to initialize the image.
-
+    ``width``×``height`` integers in range 0-9 to initialize the image:
+   
+        Image(2, 2, b'\x08\x08\x08\x08')
```
Comment (microbit-carlos) [inline docs/image.rst]: Can we also add an example with a byte array?
```
Image(2, 2, bytearray([0,1,1,0]))
```

File: docs/image.rst [hunk 6]
```diff
@@ -39,8 +50,15 @@ Classes
...
+	
+    Will display a 2 x 2 pixel image at full brightness.
+    
+    .. note::
+    
+    Keyword arguments cannot be passed to ``buffer``.
```
Comment (microbit-carlos) [inline docs/image.rst]: This line needs to be indented to be formatted as a note.

File: docs/image.rst (hunk 5 above)
Comment (microbit-mark) [reply docs/image.rst]: done in https://github.com/bbcmicrobit/micropython/pull/582/commits/24b83b74c2cf1bddd8b9c9dcbcef9096979dd24f

Comment (microbit-mark) [reply docs/image.rst]: done in https://github.com/bbcmicrobit/micropython/pull/582/commits/24b83b74c2cf1bddd8b9c9dcbcef9096979dd24f

File: docs/image.rst (hunk 6 above)
Comment (microbit-mark) [reply docs/image.rst]: done in https://github.com/bbcmicrobit/micropython/pull/582/commits/24b83b74c2cf1bddd8b9c9dcbcef9096979dd24f

File: docs/image.rst (hunk 5 above)
Comment (microbit-mark) [reply docs/image.rst]: done in https://github.com/bbcmicrobit/micropython/pull/582/commits/24b83b74c2cf1bddd8b9c9dcbcef9096979dd24f

File: docs/image.rst [hunk 7]
```diff
@@ -39,8 +51,17 @@ Classes
 
     The other form creates an empty image with ``width`` columns and
     ``height`` rows. Optionally ``buffer`` can be an array of
-    ``width``×``height`` integers in range 0-9 to initialize the image.
-
+    ``width``×``height`` integers in range 0-9 to initialize the image:
+   
+        Image(2, 2, b'\x08\x08\x08\x08')
+    or
```
Comment (microbit-carlos) [inline docs/image.rst]: Without the additional lines the formatting goes a bit wonky:

image


```suggestion

    or::

```

File: docs/image.rst [hunk 8]
```diff
@@ -39,8 +51,17 @@ Classes
 
     The other form creates an empty image with ``width`` columns and
     ``height`` rows. Optionally ``buffer`` can be an array of
-    ``width``×``height`` integers in range 0-9 to initialize the image.
-
+    ``width``×``height`` integers in range 0-9 to initialize the image:
```
Comment (microbit-carlos) [inline docs/image.rst]: Adding the extra colon to format the code line bellow in the same format as the previous code examples.
```suggestion
    ``width``×``height`` integers in range 0-9 to initialize the image::
```

**Comment by microbit-carlos** [comment]
Thank you Mark!

**Pull request merged**